streams/src/main/java/org/apache/kafka/streams/state/internals/CachingKeyValueStore.java

```diff
@@ -228,6 +232,12 @@ public void putAll(final List<KeyValue<Bytes, byte[]>> entries) {
     @Override
     public KeyValueIterator<Bytes, byte[]> range(final Bytes from,
                                                  final Bytes to) {
+        // Make sure this is a valid query
+        if (from.compareTo(to) > 0) {
+            LOG.warn("Returning empty iterator for range query with invalid range: keyFrom > keyTo.");
```

Comment:
nit: I would avoid to write variable names (i.e., `keyFrom` and `keyTo`) to a log, because they are hard to maintain consistently with the code (as you can see here).

Reply:
Ack